### PR TITLE
Removed fallback on web for drawAtlas

### PR DESCRIPTION
**Pull request by wolfenrain**
# Description

Initial testing if the fallback for the `drawAtlas` method is still needed for the web.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [ ] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.
- [ ] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [ ] I have formatted my code with `flutter format` and the `flutter analyze` does not report any problems.
- [ ] I read and followed the [Flame Style Guide].
- [ ] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [ ] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in `CHANGELOG.md`).
- [ ] No, this is *not* a breaking change.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database]. Indicate, which of these issues are resolved or fixed by this PR.*

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md

### CODE REVIEW

**Comment by luanpotter** [comment]
I tested this on chrome mobile mode on desktop.

main:
image

testing-draw-atlas-on-web:
image

I believe this means:
* both the fallback and the normal impl works on web now
* so the fallback (that that doesn't work on devices) is no longer needed
* however the normal implementation clearly shows the tile gap issue for the particular size/zoom I used. but that is a known issue unrelated to anything that the fallback just happened to fix.

**Comment by spydon** [comment]
> I tested this on chrome mobile mode on desktop.
> 
> main:
> image
> 
> testing-draw-atlas-on-web:
> image
> 
> I believe this means:
> * both the fallback and the normal impl works on web now
> * so the fallback (that that doesn't work on devices) is no longer needed
> * however the normal implementation clearly shows the tile gap issue for the particular size/zoom I used. but that is a known issue unrelated to anything that the fallback just happened to fix.

Maybe we can leave it as a selectable mode called something else then?

File: packages/flame/lib/src/sprite_batch.dart [hunk 1]
```diff
@@ -254,15 +269,18 @@ class SpriteBatch {
...
+  final Paint _emptyPaint = Paint();
+
   void render(
     Canvas canvas, {
     BlendMode? blendMode,
     Rect? cullRect,
     Paint? paint,
   }) {
-    paint ??= Paint();
+    paint ??= _emptyPaint;
```
Comment (erickzanardo) [inline packages/flame/lib/src/sprite_batch.dart:281]:
```suggestion
    paint ??= const Paint();
```
Then we don't need the field :)

**Review by erickzanardo — APPROVED** [review]

File: packages/flame/lib/src/sprite_batch.dart (hunk 1 above)
Comment (spydon) [reply packages/flame/lib/src/sprite_batch.dart:281]: It still creates a new const object right? Or does dart have a heap with const objects internally?

Comment (erickzanardo) [reply packages/flame/lib/src/sprite_batch.dart:281]: Dart will create only a single instance during the execution and use everywhere that has `const Paint()`

Comment (spydon) [reply packages/flame/lib/src/sprite_batch.dart:281]: Great, that'll look much better too :)

Comment (spydon) [reply packages/flame/lib/src/sprite_batch.dart:281]: Ah sadly `Paint` doesn't have a `const` constructor.

Comment (erickzanardo) [reply packages/flame/lib/src/sprite_batch.dart:281]: Oh no :( nvm then 😅

**Review by luanpotter — APPROVED** [review]
LGTM

**Pull request merged**